jax/_src/numpy/lax_numpy.py

```diff
@@ -1368,6 +1368,87 @@ def gradient(f, *varargs, axis: Optional[Union[int, Tuple[int, ...]]] = None,
 def isrealobj(x):
   return not iscomplexobj(x)
 
+@partial(jit, static_argnums=1)
+class _RankWarning(UserWarning):
+    """
+    Issued by `polyfit` when the Vandermonde matrix is rank deficient.
+    For more information, a way to suppress the warning, and an example of
+    `RankWarning` being issued, see `polyfit`.
+    """
+    pass
+
+@_wraps(np.polyfit)
+def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
+  _check_arraylike("polyfit", x, y, deg)
+  x = core.concrete_or_error(None, x, "The error arose in polyfit()")
+  y = core.concrete_or_error(None, y, "The error arose in polyfit()")
+  order = int(deg) + 1
```

Comment:
Should use
```
deg = core.concrete_or_error(int, deg, msg)
order = deg + 1
```
`deg` and `order` must be concrete, because their values are used in logical statements and to control the shapes of arrays.